drivers: intel_gna: Intel GNA Neural Network Accelerator Driver

sathishkuttan opened the pull request:
drivers: intel_gna: Intel GNA Neural Network Accelerator Driver

This PR adds APIs and a driver implementation for the Intel GNA neural network accelerator in Intel S1000 ASIC.

Commit:
drivers: intel_gna: Add APIs for Intel GNA

APIs for Intel's GNA Low Power Neural Network Accelerator

Signed-off-by: Sathish Kuttan <sathish.k.kuttan@intel.com>

Commit:
kconfig: intel_s1000: Add neural_net Kconfig

Added Kconfig files for Intel neural network accelerator
(GNA) driver

Signed-off-by: Sathish Kuttan <sathish.k.kuttan@intel.com>

Commit:
drivers: neural_net: Add cmake files

Added cmake files for neural_net acceleration drivers

Signed-off-by: Sathish Kuttan <sathish.k.kuttan@intel.com>

Commit:
drivers: intel_gna: Driver for Intel GNA

Driver implementation for Intel GNA Neural Network Accelerator

Signed-off-by: Sathish Kuttan <sathish.k.kuttan@intel.com>

nashif commented on include/gna.h:
```diff
@@ -0,0 +1,160 @@
+/**
+ * @endcond
+ */
+
```
You will need to document the API with doxygen decorations.

sathishkuttan replied:
Good catch. Done.

Commit:
soc: intel_s1000: GNA model - linker script entry

Add a linker script entry for embedding a GNA model at a
4KB page aligned address in .data section

Signed-off-by: Sathish Kuttan <sathish.k.kuttan@intel.com>

pfalcon commented on boards/xtensa/intel_s1000_crb/intel_s1000_crb_defconfig:
```diff
@@ -54,3 +54,6 @@ CONFIG_AUDIO_CODEC=y
 CONFIG_AUDIO_DMIC=y
 CONFIG_AUDIO_INTEL_DMIC=y
+
+CONFIG_NEURAL_NET=y
```
IMHO, that's too generic. If this is an option to enable neural net **accelerator** **drivers**, then the most important thing about it is that it's *hardware driver*, and that's completely not mirrored in the option name.

sathishkuttan replied:
How about `CONFIG_NEURAL_NET_ACCEL`

pfalcon replied:
Definitely better, +1 from me.

sathishkuttan replied:
Thanks. Updated to `CONFIG_NEURAL_NET_ACCEL`.

pfalcon replied:
Thanks too! So, I'm not qualified to review the rest of the patch to approve it, but informally, +1, it's great to see such cool features added to Zephyr.
